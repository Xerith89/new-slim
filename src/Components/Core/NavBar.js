import React from 'react'
import {Component} from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faCog } from '@fortawesome/free-solid-svg-icons'
import DropDown from './DropDown'

export default class NavBar extends Component {
    
    constructor(props) {
        super(props)

        this.state = {
            optionsOpen:false,
        }
    }

    toggleDropDown = () => {
        this.setState({
            optionsOpen: !this.state.optionsOpen,
        })
    }
    
    render() {

        const optionsIsOpen = this.state.optionsOpen;
        let menu;

        if (optionsIsOpen) {
            menu = <DropDown/>
        }

        return (
           <div data-testid="nav">
            <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
                <h1 className="navbar-brand text-white" >New SLIM</h1>
                <div className="collapse navbar-collapse" id="navbarSupportedContent">
                    <ul className="navbar-nav ml-auto">
                        <li className="nav-item nav-link active">
                            <h3>Welcome, Username</h3>
                        </li>
                        <div className="dropdown">
                            <li className="nav-item">
                                <button className="nav-link bg-transparent border-0" data-testid="toggler" onClick={this.toggleDropDown} ><FontAwesomeIcon icon={faCog} size="2x" /></button>
                            </li>
                            {menu}
                        </div>
                    </ul>
                </div>
            </nav>
        </div>
        )
    }
}
