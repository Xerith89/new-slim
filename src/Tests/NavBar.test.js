import React from 'react';
import NavBar from '../Components/Core/NavBar';
import {render} from '@testing-library/react';

test('NavBar component renders to the DOM', () => {
    const {getByTestId} = render(<NavBar />);
    expect(getByTestId('nav')).toMatchSnapshot();
});
