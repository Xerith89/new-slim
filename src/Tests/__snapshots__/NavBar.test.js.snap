// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component renders to the DOM 1`] = `
<div
  data-testid="nav"
>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      class="navbar-brand text-white"
    >
      New SLIM
    </a>
    <div
      class="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        class="navbar-nav mr-auto"
      >
        <li
          class="nav-item active"
        >
          <a
            class="nav-link"
            href="#"
          >
            Home 
            <span
              class="sr-only"
            >
              (current)
            </span>
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#"
          >
            Link
          </a>
        </li>
        <li
          class="nav-item dropdown"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            class="nav-link dropdown-toggle"
            data-toggle="dropdown"
            href="#"
            id="navbarDropdown"
            role="button"
          >
            Dropdown
          </a>
          <div
            aria-labelledby="navbarDropdown"
            class="dropdown-menu"
          >
            <a
              class="dropdown-item"
              href="#"
            >
              Action
            </a>
            <a
              class="dropdown-item"
              href="#"
            >
              Another action
            </a>
            <div
              class="dropdown-divider"
            />
            <a
              class="dropdown-item"
              href="#"
            >
              Something else here
            </a>
          </div>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link disabled"
            href="#"
          >
            Disabled
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
